soccer/vision/ball/CameraBall.cpp

```diff
@@ -0,0 +1,34 @@
+#include "CameraBall.hpp"
+
+RJ::Time CameraBall::getTimeCaptured() const {
+    return timeCaptured;
+}
+
+Geometry2d::Point CameraBall::getPos() const {
+    return pos;
+}
+
+CameraBall CameraBall::CombineBalls(const std::list<CameraBall>& balls) {
```

Comment:
Why does this take a `std::list` instead of a front/back iterator?

Reply:
I'll be honest, it's more probably personal preference. I started in C98 so this feels better to me. I also can't say that I've seen people pass iterators around instead of the list itself in anything except the standard libraries and things that are operations directly on lists (sorting, search etc).

Reply:
The only reason I ask is that you do a lot of copy operations later to turn vectors into lists so you can pass them to this function, which isn't great. Alternatively I suppose you could pass in a reference to a collection of arbitrary type via templates, but iterators are more idiomatic.

Reply:
I'll probably just swap them over to vectors to match. This is part of the whole copy fiasco that I forgot about

Reply:
It looks like they're lists for the sake of arbitrary removal from the container, right? (although now that I think about it, vectors are still probably better for any sane estimate count - I suppose if big-O is coming into play here something is _seriously_ wrong).

Reply:
Yeah. You can only take in iterators if you have a templated library/functions. Vector is faster in most every use case of reasonable sizes tho!!!

Reply:
Yeah. You can only take in iterators if you have a templated library/functions. Vector is faster in most every use case of reasonable sizes tho!!!

Reply:
Yeah. You can only take in iterators if you have a templated library/functions. Vector is faster in most every use case of reasonable sizes tho!!!

Reply:
Yeah. You can only take in iterators if you have a templated library/functions. Vector is faster in most every use case of reasonable sizes tho!!!

Reply:
Yeah. You can only take in iterators if you have a templated library/functions. Vector is faster in most every use case of reasonable sizes tho!!!

Reply:
Yeah. You can only take in iterators if you have a templated library/functions. Vector is faster in most every use case of reasonable sizes tho!!!

Reply:
Yeah. You can only take in iterators if you have a templated library/functions. Vector is faster in most every use case of reasonable sizes tho!!!